docs/guides/waitlist.mdx

```diff
@@ -0,0 +1,154 @@
+  import './globals.css'
+
+  export default function RootLayout({ children }: { children: React.ReactNode }) {
+    return (
+      <ClerkProvider waitlistUrl="/waitlist">
+        <html lang="en">
+          <body>{children}</body>
+        </html>
+      </ClerkProvider>
+    )
+  }
+  ```
+
+  ### Add the `<Waitlist />` component
+
+  The [`<Waitlist />`](/docs/components/waitlist/overview) component renders a form that allows users to join for early access to your app.
+
+  The following example includes a basic implementation of the `<Waitlist />` component hosted on the `/` route (the home page). You can use this as a starting point for your own implementation.
+
+  ```jsx {{ filename: '/app/page.tsx' }}
+  import { Waitlist } from '@clerk/nextjs'
+
+  export default function Page() {
+    return <Waitlist />
+  }
+  ```
+
+  ### Add sign-in functionality
+
+  To allow users to sign in once they've been approved from the waitlist, you must:
+
+  - [Add `clerkMiddleware()` to your app.](#add-clerkmiddleware-to-your-app)
+  - [Add a sign-in page.](#add-a-sign-in-page)
+
+  #### Add `clerkMiddleware()` to your app
+
+  [`clerkMiddleware()`](/docs/references/nextjs/clerk-middleware#clerk-middleware) grants you access to user authentication state throughout your app, on any route or page. It also allows you to protect specific routes from unauthenticated users. To add `clerkMiddleware()` to your app, follow these steps:
+
+  1. Create a `middleware.ts` file.
+
+  - If you're using the `/src` directory, create `middleware.ts` in the `/src` directory.
+  - If you're not using the `/src` directory, create `middleware.ts` in the root directory alongside `.env.local`.
+
+  1. In your `middleware.ts` file, export the `clerkMiddleware()` helper:
+
+  ```tsx {{ filename: 'middleware.ts' }}
+  import { clerkMiddleware } from '@clerk/nextjs/server'
+
+  export default clerkMiddleware()
+
+  export const config = {
+    matcher: [
+      // Skip Next.js internals and all static files, unless found in search params
+      '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
+      // Always run for API routes
+      '/(api|trpc)(.*)',
+    ],
+  }
+  ```
+
+  1. By default, `clerkMiddleware()` will not protect any routes. All routes are public and you must opt-in to protection for routes. See the [`clerkMiddleware()` reference](/docs/references/nextjs/clerk-middleware) to learn how to require authentication for specific routes.
+
+  #### Add a sign-in page
+
+  The following example demonstrates how to render the `<SignIn />` component. **Pass the `waitlistUrl` prop to the component to provide the target of the 'Waitlist' link that's rendered in the `<SignIn />` component.**
+
+  ```tsx {{ filename: 'app/sign-in/[[...sign-in]]/page.tsx' }}
+  import { SignIn } from '@clerk/nextjs'
+
+  export default function Page() {
+    return <SignIn waitlistUrl="/" />
```

Comment:
If we pass the `waitlistUrl` in `<ClerkProvider>` we don't need to add it here

Reply:
addressed: [a93f692](https://github.com/clerk/clerk-docs/pull/1734/commits/a93f692050989406698140b8968afd82161215e8)